janitor/functions/conditional_join.py

```diff
@@ -300,6 +276,11 @@ def _conditional_join_preliminary_checks(
         check_column(right, right_on)
         _check_operator(op)
 
+    if all(
+        (op == _JoinOperator.STRICTLY_EQUAL.value for *_, op in conditions)
+    ):
+        raise ValueError("""Equality only joins are not supported.""")
```

Comment:
Looks like codecov is complaining that this line wasn't covered by the tests.

Can we add a test for this case, please?